Python/pywarpx/_libwarpx.py

```diff
@@ -561,6 +563,53 @@ def get_particle_arrays(species_name, comp_name, level):
     _libc.free(data)
     return particle_data
 
+def get_particle_boundary_buffer(species_name, boundary, comp_name, level):
+    '''
+
+    This returns a list of numpy arrays containing the particle array data
+    for a species that has been scraped by a specific simulation boundary.
+
+    The data for the numpy arrays are not copied, but share the underlying
+    memory buffer with WarpX. The numpy arrays are fully writeable.
+
+    Parameters
+    ----------
+
+        species_name   : the species name that the data will be returned for.
+        boundary       : the boundary from which to get the scraped particle data.
+                         The formula for the boundary is dimension * 2 + side.
+                         Where dimension is 1, 2, or 3, and side is 0 or 1, representing
+                         the low, or high boundary.
```

Comment:
Oh cool. So actually we can change this to take a string argument of the form `z_hi` for example?

Reply:
Yep I think this is easy to do